src/momento/simple_cache_client_async.py

```diff
@@ -0,0 +1,371 @@
+        print("arch -x86_64 {} {}".format(sys.executable, *sys.argv), file=sys.stderr)
+        print("-".join("" for _ in range(99)), file=sys.stderr)
+    raise e
+
+from momento.responses import (
+    CacheDeleteResponseBase,
+    CacheGetResponseBase,
+    CacheSetResponseBase,
+    CreateCacheResponseBase,
+    CreateSigningKeyResponse,
+    DeleteCacheResponseBase,
+    ListCachesResponseBase,
+    ListSigningKeysResponse,
+    RevokeSigningKeyResponse,
+)
+
+
+class SimpleCacheClientAsync:
+    """Async Simple Cache Client"""
+
+    # For high load, we might get better performance with multiple clients, because the server is
+    # configured to allow a max of 100 streams per connection.  In the javascript SDK, multiple
+    # clients resulted in an obvious performance improvement.  However, in the python SDK I have
+    # not yet been able to observe a clear benefit.  So for now, we are putting the plumbing in
+    # place so that we can more easily test performance with multiple connections in the future,
+    # but we are leaving the default value set to 1.
+    #
+    # We are hard-coding the value for now, because we haven't yet designed the API for
+    # users to use to configure tunables:
+    # https://github.com/momentohq/dev-eco-issue-tracker/issues/85
+    _NUM_CLIENTS = 1
+
+    def __init__(self, configuration: Configuration, credential_provider: CredentialProvider, default_ttl: timedelta):
+        """Creates an async SimpleCacheClient
+
+        Args:
+            configuration (Configuration): An object holding configuration settings for communication with the server.
+            credential_provider (CredentialProvider): An object holding the auth token and endpoint information.
+            default_ttl (timedelta): A default Time To Live timedelta for cache objects created by this client.
+                It is possible to override this setting when calling the set method.
+        Raises:
+            IllegalArgumentException: If method arguments fail validations.
+        """
+        _validate_request_timeout(configuration.get_transport_strategy().get_grpc_configuration().get_deadline())
+        self._logger = logs.logger
+        self._next_client_index = 0
+        self._control_client = _ScsControlClient(credential_provider)
+        self._data_clients = [
+            _ScsDataClient(configuration, credential_provider, default_ttl)
+            for _ in range(SimpleCacheClientAsync._NUM_CLIENTS)
+        ]
+
+    async def __aenter__(self) -> "SimpleCacheClientAsync":
+        return self
+
+    async def __aexit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc_value: Optional[BaseException],
+        traceback: Optional[TracebackType],
+    ) -> None:
+        await self._control_client.close()
+        for data_client in self._data_clients:
+            await data_client.close()
+
+    async def create_cache(self, cache_name: str) -> CreateCacheResponseBase:
```

Comment:
just one naming nit here (and in all of the other public APIs) - devs who are using mypy will see the return type for all the APIs as including the word `Base`, which feels maybe a little weird to me?  Maybe it's okay.

In the JS SDK we decided to use names like `CreateCacheResponse` and `CreateCache.Success` so that the word `Base` wouldn't leak out into the user-facing API.  I think I still have a slight preference for that but you could convince me to keep it this way.

Reply:
That sounds better to me. Seeing `Base` here is a bit jarring for me.

Reply:
Fixed in https://github.com/momentohq/client-sdk-python/pull/179/commits/4fe0b426c4268f0d137f305d62d90ec498119bef